src/components/projects-section/projects-carousel/index.tsx

```diff
@@ -0,0 +1,140 @@
+"use client";
+import styles from "./styles.module.css";
+import { AnimatePresence, motion, MotionConfig } from "framer-motion";
+import { mock_projects_buttons, mock_projects_data } from "@/mock-data/projects";
+import { FaChevronLeft, FaChevronRight } from "react-icons/fa6";
+import { Slide } from "./slide/slide";
+import { useState } from "react";
+import { ProjectsIcon } from "../projects-icon";
+
+export const ProjectsCarousel = () => {
+  const [activeProjectIndex, setActiveProjectIndex] = useState(0);
+  const [direction, setDirection] = useState<"left" | "right">("right");
+
+  const handlePreviousSlide = () => {
+    setDirection("left");
+    if (activeProjectIndex === 0) return setActiveProjectIndex(mock_projects_data.length - 1);
+
+    return setActiveProjectIndex((prev) => prev - 1);
+  };
+
+  const handleNextSlide = () => {
+    setDirection("right");
+    if (activeProjectIndex === mock_projects_data.length - 1) {
+      return setActiveProjectIndex(0);
+    }
+
+    return setActiveProjectIndex((prev) => prev + 1);
+  };
+  const motionInitial = direction === "right" ? "hiddenRight" : "hiddenLeft";
+  const slideVariants = {
+    hiddenRight: {
+      x: "100%",
+      opacity: 0,
+    },
+    hiddenLeft: {
+      x: "100%",
+      opacity: 0,
+    },
+    visible: {
+      x: "0",
+      opacity: 1,
+      transition: {
+        duration: 0.8,
+      },
+    },
+    exit: {
+      opacity: 0,
+      x: direction === "right" ? "-100%" : "100%",
+      transition: {
+        duration: 0.4,
+      },
+    },
+  };
+
+  return (
+    <>
+      <ul className={styles.projects_list_button}>
+        {mock_projects_buttons.map((button, index) => (
+          <li
+            key={button.title}
+            title={button.title}
+          >
+            <button
+              className={styles.project_button}
+              onClick={() => {
+                setActiveProjectIndex(index);
```

Comment:
Nessa linha, podemos adicionar uma verificação. 
```jsx
if (activeProjectIndex === index) return;
```
O motivo é que estamos setando um estado direto nesse botão, cada vez que o usuário clica no botão temos o re-render do componente todo. Adicionando essa verificação, evitamos o spam e não afetamos o funcionamento. O que acha?

Reply:
@GiovaniBiagi Seria pra so fazer o set caso va mudar o activeProjectIndex? Se for isso, eh uma otima, nao tinha pensado nisso. 
O que acha de passarmos disabled pro botao, assim a gente nem chama a funcao

Reply:
Exatamente @marcelluscaio ! Podemos fazer isso, também resolve.

Reply:
Curto muito fazer isso, especialmente em useEffect, agora nesse caso aqui, se não me engano, o setState não re-renderiza se o valor que a gente passar for o atual (que é essa verificação que a gente faria), porque não geraria nenhum diff no Virtual DOM.

Perfilei ele aqui no React Profiler e confirmei esse comportamento, mesmo, se eu clicar no botão do slide ativo ele não re-renderiza.

E achei esse trecho da [documentação aqui do React](https://react.dev/reference/react/useState#setstate-caveats) que também fala desse tópico.

Mas não sei se deixei passar alguma outra coisa

Reply:
Então, sem necessidade de alterações aqui! Valeu @pdrmenezes !